Exercise 2

jackt24 opened the pull request:
I wasnt able to complete the feature where you update the status of a to do item (i could only add a button) 

I also need a bit of help understanding part 5.

If you could arrange some time with me for these I would be greatful as I am already a little behind.

BenRamchandani reviewed (commented):
Hi Jack,

I've left some comments below to help with moving cards to the "complete" status.

For part 5 the idea is to create an object to represent an Item rather than using a dict like `{id: 123, name: 'Item name'}`.
As an example a Person class might look like:
```python
class Person:
    def __init__(self, name)
        self.name = name

# Use like this
myPerson = Person('Ben')
print(myPerson.name)
```
You should define an `Item` class with the properties you want and change `get_items` to return a list of them.

If you'd like more help you can ask questions in Slack/by email, or book in a support call, the links are in the `#support-call-links` channel in Slack.

jackt24 commented:
@BenRamchandani Thanks for all your comments, really helpful. I have addressed all the issues and I believe I have completed all 5 tasks so please take another look when possible.

BenRamchandani commented on README.md:
```diff
@@ -32,6 +32,14 @@ $ cp .env.template .env  # (first time only)
 
+You will need to setup a Trello account, create a ToDo and a Complete board and change the following keys in the .env file:
+
+SECRET_KEY = Key generated on the Trello site
+TOKEN = Token generated on the trello site
+BOARD = Id of the Board
+LIST = Id of the Todo list
+COMPLETELIST = Id of the complete list
```
Would be good to add these to .env.template

BenRamchandani commented on todo_app/app.py:
```diff
@@ -1,28 +1,30 @@
-# I added add functionality to the route rather than a seperate page as this seemed more efficient
-@app.route('/', methods=['POST', 'GET'])
+# Populate lists and add items 
+@app.route('/',methods=['POST', 'GET'])
```
I think the comment is out of date now you have a separate `/add` endpoint, and you don't need to allow POST requests

BenRamchandani commented on todo_app/data/session_items.py:
```diff
@@ -1,69 +1,108 @@
-
-    Returns:
-        item: The saved item, or None if no items match the specified ID.
-    """
     items = get_items()
     return next((item for item in items if item['id'] == int(id)), None)
```
I don't think this is used any more

BenRamchandani commented on todo_app/data/session_items.py:
```diff
@@ -1,69 +1,108 @@
+import requests, os
+from dotenv import load_dotenv
+from flask.helpers import url_for
+from todo_app import item
```
Could do `from todo_app.item import Item` if you don't want to have to do `item.Item` to use the class

BenRamchandani commented on README.md:
```diff
@@ -32,6 +32,14 @@ $ cp .env.template .env  # (first time only)
 
+You will need to setup a Trello account, create a ToDo and a Complete board and change the following keys in the .env file:
+
+SECRET_KEY = Key generated on the Trello site
```
Flask uses `SECRET_KEY` to sign session cookies (that's why you needed `SECRET_KEY` set for exercise 1), we're not using session storage any more so it works fine to use it for something else, but just to avoid confusion it might be better to name it something else, e.g. `TRELLO_KEY`

BenRamchandani reviewed (commented):
Thanks for doing the markups 🙂 I've left some comments on things you could tidy up, but the changes all look good.

The Trello Token you committed earlier is still valid though! Please go to <https://trello.com/>, click on your profile in the top right, then go to settings -> Applications and revoke the token (probably called "Server Token").  
Then you can make a new Token by going to <https://trello.com/app-key> and clicking on the link to "manually generate a token" as before.

jackt24 commented:
@BenRamchandani thanks for all the feedback! I have implemented and will merge when approved.

BenRamchandani reviewed (approved):
Looks great, thanks for doing all the markups 🙂 Merge when you're ready.
